File: backend/app/Models/GambarKamar.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class GambarKamar extends Model
{
    protected $table = 'gambar_kamar';
    protected $primaryKey = 'id_gambarkamar';

    protected $fillable = [
        'id_kamar',
        'nama_gambarkamar',
        'keterangan_gambarkamar',
        'url_gambarkamar',
    ];

    public function kamar()
    {
        return $this->belongsTo(Kamar::class, 'id_kamar', 'id_kamar');
    }
}

File: frontend/lib/entity/kamar/gambar_kamar.dart
class GambarKamar {
  final int idGambarKamar;
  final int idKamar;
  final String namaGambarKamar;
  final String keteranganGambarKamar;
  final String urlGambarKamar;

  GambarKamar({
    required this.idGambarKamar,
    required this.idKamar,
    required this.namaGambarKamar,
    required this.keteranganGambarKamar,
    required this.urlGambarKamar,
  });

  factory GambarKamar.fromJson(Map<String, dynamic> json) {
    return GambarKamar(
      idGambarKamar: json['id_gambarkamar'] ?? 0,
      idKamar: json['id_kamar'] ?? 0,
      namaGambarKamar: json['nama_gambarkamar'] ?? '',
      keteranganGambarKamar: json['keterangan_gambarkamar'] ?? '',
      urlGambarKamar: json['url_gambarkamar'] ?? '',
    );
  }
}
